### PR TITLE
Port validate_drp algorithms

### PR DESCRIPTION
**Pull request by bechtol**
Opening PR to port validate_drp algorithms to new_framework. The comments on the JIRA ticket provide a summary of the functions that need to be ported over.

https://jira.lsstcorp.org/browse/DM-25504

In this initial implementation, I opted to port the functions into three files in the `python/metric_pipeline_utils` folder: `phot_repeat.py` (already existing), `coord_util.py` (new), `tex.py` (new). Below is a summary of where the functions have been moved.

- repeatability (--> `phot_repeat.py`)
  - calcPhotRepeat
  - calcPhotRepeatSample
  - computeWidths
  - getRandomDiffRmsInMmags
  - getRandomDiff
- util
  - averageRaFromCat (--> `coord_util.py`)
  - averageDecFromCat (--> `coord_util.py`)
  - sphDist (--> `coord_util.py`)
  - averageRaDecFromCat (--> `coord_util.py`)
  - averageRaDec (--> `coord_util.py`)
  - medianEllipticity1ResidualsFromCat (--> `tex.py`)
  - medianEllipticity2ResidualsFromCat (--> `tex.py`)
- calcsrd.tex (--> `tex.py`)
  - correlation_function_ellipticity_from_matches
  - select_bin_from_corr
  - correlation_function_ellipticity

In this first commit, I tried to make minimal changes to the validate_drp functions, including doc strings. In a future commit for the PR, it probably makes sense to go through the doc strings at least and clean up references to validate_drp.

### CODE REVIEW

File: python/metric_pipeline_utils/coord_util.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    dec : `numpy.array` [`float`]
+        Array of Dec in radians.
+    Notes
+    -----
+    Uses the Haversine formula to preserve accuracy at small angles.
+    Law of cosines approach doesn't work well for the typically very small
+    differences that we're looking at here.
+    """
+    # Haversine
+    dra = ra - ra_mean
```
Comment (SimonKrughoff) [inline python/metric_pipeline_utils/coord_util.py:105]: I believe the `geom` code in the stack uses the Haversine formula.  It's a little unfortunate that we reimplement it here.  We don't need to change it here, but I'd like an issue in the non-PoC epic to track it.

**Review by SimonKrughoff — CHANGES_REQUESTED** [review]
The porting of code looks good, but I think we need tests for all the functions in the utils area.

I'm open for discussion on whether that happens in an independent PR or not.

**Comment by bechtol** [comment]
I think this is ready for another round of review. I've verified running scons on lsst-devl03.

**Review by SimonKrughoff — APPROVED** [review]
Thanks very much for doing this.  It looks good to me.  We may think about using `setUp` and `tearDown` to reduce some duplication, but that's not important here, and I know the tests I wrote could benefit from that simplification too.

**Pull request merged**